ml_tools/writers.py

```diff
@@ -367,3 +480,20 @@ def flush(self):
 
     def close(self):
         self.flush()
+
+def _resolve_write_fn(data):
```

Comment:
Can we also store a dict of numpy arrays as a single npz file? I think it would be useful to not have to create a different file for each array in a dict.